packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -388,7 +388,7 @@ class KotlinGenerator extends StructuredGenerator<KotlinOptions> {
                 final String forceUnwrap =
                     func.returnType.isNullable ? '?' : '';
                 indent.writeln(
-                    'val result = if (it is Int) it.toLong() else it as$forceUnwrap Long');
```

Comment:
`forceUnwrap` isn't the right name for this now. I would recommend moving the logic into `_cast`, adding an argument for nullability of the target type, since I believe we have other places where we're doing this '?'/'' logic that would be better consolidated.